### ISSUE TITLE
No MessageAttributes on SQS message (send and received directly with SQS)

### ISSUE THREAD

**Issue by maccamlcQ** [localstack/localstack#1857]
Using 0.10.4 docker image

I noticed that when I am testing a service that sends message to a SQS queue, they are received without any message attributes.

The message attributes seem to work in the SNS -> SQS case, but not SQS directly.

I see attributes, fwiw. Just not the custom MessageAttributes :)

**Comment by Mamoru1234** [comment]
Is this issue related to #1520 ?
Have you tried latest version of localstack?
Could you please provide a test?
Thanks and regards :)

**Comment by maccamlcQ** [comment]
Thanks for the response. I just tried with 0.10.7 but still experienced the same issue.

I don't think it is related, as this is in regards to custom message attributes, not the system queue attributes.

I created a minimal test using localstack-utils that fails due to empty MessageAttributes on the received message.

```
import cloud.localstack.TestUtils;
import cloud.localstack.docker.LocalstackDockerExtension;
import cloud.localstack.docker.annotation.LocalstackDockerProperties;
import com.amazonaws.services.sqs.AmazonSQSAsync;
import com.amazonaws.services.sqs.model.CreateQueueResult;
import com.amazonaws.services.sqs.model.Message;
import com.amazonaws.services.sqs.model.MessageAttributeValue;
import com.amazonaws.services.sqs.model.ReceiveMessageResult;
import com.amazonaws.services.sqs.model.SendMessageRequest;
import com.amazonaws.services.sqs.model.SendMessageResult;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import java.util.List;
import java.util.Optional;

import static org.assertj.core.api.Assertions.assertThat;

@LocalstackDockerProperties(imageTag = "0.10.7", services = {"sqs"}, randomizePorts = true)
@ExtendWith(LocalstackDockerExtension.class)
public class Example {

    @Test
    void exampleTest() {
        final AmazonSQSAsync sqsAsync = TestUtils.getClientSQSAsync();

        final CreateQueueResult myqueue = sqsAsync.createQueue("myqueue");

        final SendMessageResult sendMessageResult = sqsAsync.sendMessage(
                new SendMessageRequest()
                        .withQueueUrl(myqueue.getQueueUrl())
                        .addMessageAttributesEntry("testKey", new MessageAttributeValue().withStringValue("a value to see").withDataType("String"))
                        .withMessageBody("Simple body")
        );

        final String messageId = sendMessageResult.getMessageId();

        final ReceiveMessageResult receiveMessageResult = sqsAsync.receiveMessage(myqueue.getQueueUrl());

        final List<Message> messages = receiveMessageResult.getMessages();

        final Optional<Message> messageOptional = messages.stream()
                .filter(message -> messageId.equals(message.getMessageId()))
                .findFirst();

        assertThat(messageOptional)
                .hasValueSatisfying(message -> {
                    assertThat(message.getBody()).isEqualTo("Simple body");
                    assertThat(message.getMessageAttributes())
                            .isNotEmpty() // <-- FAILS HERE
                            .hasEntrySatisfying("testKey", value -> {
                                assertThat(value.getDataType()).isEqualTo("String");
                                assertThat(value.getStringValue()).isEqualTo("a value to see");
                            });
                });
    }

}
```

Hope that helps

**Comment by bmastalski** [comment]
I'm experiencing the same issue on 0.10.7.

**Comment by whummer** [comment]
Thanks for reporting the example @maccamlcQ . We've added a test in #1955 . This seems to be working if you add the message attribute wildcard "All" to the retrieval of the message:
```
final ReceiveMessageRequest request = new ReceiveMessageRequest(myqueue.getQueueUrl()).
            withMessageAttributeNames("All");
final ReceiveMessageResult receiveMessageResult = sqsAsync.receiveMessage(request);
```
Did you have a chance to test this against real AWS - are the message attributes returned in AWS if you don't specify the message attribute names? Thanks

**Comment by bmastalski** [comment]
Thanks @whummer !

I can confirm that the following works with boto3 and localstack 0.10.7
`sqs.Queue.receive_messages(MessageAttributeNames=[".*"])`.

I have also checked what the default AWS behaviour is. When you don't specify the `MessageAttributeNames` argument, it doesn't return them. There is no issue with localstack after all.

On the other hand, moto returns message attributes by default, which was the source of my confusion when I started using localstack in my tests.
